### PR TITLE
feat(infinity): add a baseline implementation for CCO2 support

### PR DESCRIPTION
**Pull request by must-be-carbon**

### CODE REVIEW

File: src/infinity/libraries/TokenSwap/LibUniswapV2Swap.sol [hunk 1]
```diff
@@ -16,15 +17,29 @@ library LibUniswapV2Swap {
...
+            amountsOut = IUniswapV2Router02(router).swapTokensForExactTokens(
+                amountOut,
+                amountIn,
+                path,
+                address(this),
+                block.timestamp
+            );
+        } else {
+            IUniswapV2Router02(router).swapTokensForExactTokens(
+                amountOut,
```
Comment (psparacino) [inline src/infinity/libraries/TokenSwap/LibUniswapV2Swap.sol:36]: should the else condition call a different function?

File: src/infinity/libraries/Bridges/LibCoorestCarbon.sol [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    function getSpecificRetirementFee(address carbonToken, uint256 amount) external view returns (uint256 feeAmount) {
+        uint256 retireAmount = amount;
+        require(retireAmount > 0, "Retire amount should be greater than 0");
+
+        FeeParams memory feeParams = getFeePercent(carbonToken);
+
+        feeAmount =
+            ((retireAmount * feeParams.feeRetireDivider) / (feeParams.feeRetireDivider - feeParams.feeRetireBp)) -
+            retireAmount;
+    }
```
Comment (psparacino) [inline src/infinity/libraries/Bridges/LibCoorestCarbon.sol:109]: in the cco2 contract the fee is calculated:
```solidity
     uint256 amountForCoorest = _amount * mintingPercentage / decimalRatio ;
```

the calculation here will result in a slightly larger feeAmount, correct? Will that cause a discrepancy between the retired amount emitted by the RA and the retired amount emitted by cco2?

File: src/infinity/libraries/TokenSwap/LibUniswapV2Swap.sol (hunk 1 above)
Comment (must-be-carbon) [reply src/infinity/libraries/TokenSwap/LibUniswapV2Swap.sol:39]: That's sorted now. 
Can you please check?

File: src/infinity/libraries/Bridges/LibCoorestCarbon.sol [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    function retireCarbonToken(
+        address carbonToken,
+        uint256 retireAmount,
+        LibRetire.RetireDetails memory details
+    ) internal returns (uint256 poccId) {
+        require(details.beneficiaryAddress != address(0), "Beneficiary Address can't be 0");
+
+        IERC20(carbonToken).approve(C.coorestPool(), retireAmount);
+
+        // Transfer PoCC to beneficiary that's minted in favor RA
```
[the remark this answers is not in the document]
Comment (must-be-carbon) [reply src/infinity/libraries/Bridges/LibCoorestCarbon.sol:59]: Absolutely.

**Review by cujowolf — APPROVED** [review]

**Pull request merged**